### PR TITLE
chore: Automatic exe signing and release 

### PR DESCRIPTION
**Pull request by v1v**
## Summary

Sign windows binaries

### example release 
https://github.com/v1v/synthetics-recorder/releases/tag/untagged-f5a09d0327a2708aba7c

The exe file doesn't cause warning messages on a Windows VM.

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -181,12 +190,42 @@ pipeline {
...
+            deleteDir()
+            // probably a zip file is required to be Unified Release compliance.
+            git(credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba', url: 'git@github.com:elastic/infra.git')
+            bat 'dir'
+            dir('cd\\release\\windows-artifact-signer') {
+              bat 'dir'
+              unstash('exe')
+              bat 'dir'
+              bat("cmd /c '.\\ci.bat' sign --cleanup --product Elastic_Synthetics_Recorder --input dist\\Elastic_Synthetics_Recorder.zip")
+              // TODO: stash windows signed artifact
```
Comment (v1v) [inline .ci/Jenkinsfile:214]:
```suggestion
              dir('dist') {
                bat 'dir'
              }
              // TODO: stash windows signed artifact
```

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -181,12 +190,47 @@ pipeline {
...
+          agent { label 'cloud-hsm' }
+          options { skipDefaultCheckout() }
+          steps {
+            deleteDir()
+            // probably a zip file is required to be Unified Release compliance.
+            git(credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba', url: 'git@github.com:elastic/infra.git')
+            dir('cd\\release\\windows-artifact-signer') {
+              unstash('exe')
+              dir('dist') {
+                zip(glob: '*.exe', zipFile: 'Elastic_Synthetics_Recorder.zip')
```
Comment (v1v) [inline .ci/Jenkinsfile:211]:
```suggestion
                // in order to use the Unified Release, it's required a zip file with a particular naming convention and also the content of the zip file to be aligned with that naming convention
                def files = findFiles(glob: '*.exe')
                zipName = files[0].name
                bat "ren ${zipName} Elastic_Synthetics_Recorder.exe"
                bat 'dir'
                zip(glob: '*.exe', zipFile: '${zipName}.zip')
                bat 'dir'
```

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -155,6 +155,7 @@ pipeline {
         }
       }
       environment {
+        TAG_NAME = "${env.BRANCH_NAME}"
```
Comment (v1v) [inline .ci/Jenkinsfile:158]: Only for testing purposes, so it's not required
```suggestion
```

File: .ci/Jenkinsfile [hunk 4]
```diff
@@ -181,12 +180,45 @@ pipeline {
...
+          options { skipDefaultCheckout() }
+          environment {
+            BASE_DIR_WIN = "${BASE_DIR.replaceAll('\\/', '\\\\')}"
+          }
+          steps {
+            deleteDir()
+            unstash 'source'
+            unstash('exe')
+            bat(label: 'debug dir', script: 'dir')
+
```
Comment (v1v) [inline .ci/Jenkinsfile:202]: remove debug entries?
```suggestion
```

File: .ci/Jenkinsfile [hunk 5]
```diff
@@ -279,3 +314,33 @@ def runReleaseWithKeychain() {
...
+def withAzureSigningCreds(Closure body) {
+  def syntheticsRecorderWindowsVault = 'secret/observability-team/ci/synthetics-recorder-windows-vault'
+  def vault = getVaultSecret(secret: syntheticsRecorderWindowsVault).data
+  if (!vault.containsKey('vault_secret_id') || !vault.containsKey('vault_role_id') || !vault.containsKey('vault_addr')) {
+    error("withAzureSigningCreds: ${syntheticsRecorderWindowsVault} secert must contain vault_secret_id, vault_role_id and vault_addr entries")
+  }
+  def token = getVaultSecret.getVaultToken(vault.vault_addr, vault.vault_role_id, vault.vault_secret_id)  
+  def cloudHsmPath = 'release/cloud-hsm'
+  def azureSecrets = getVaultSecret.getVaultSecretObject(vault.vault_addr, 'secret/release/cloud-hsm', token).data
```
Comment (v1v) [inline .ci/Jenkinsfile:326]:
```suggestion
  def cloudHsmPath = 'secret/release/cloud-hsm'
  def azureSecrets = getVaultSecret.getVaultSecretObject(vault.vault_addr, cloudHsmPath, token).data
```

File: .ci/scripts/create-github-release.sh [hunk 6]
```diff
@@ -19,5 +19,5 @@ gh release \
   "${TAG_NAME}" \
   --draft \
   --title "${TAG_NAME}" \
-  --repo "elastic/${REPO}" \
+  --repo "v1v/${REPO}" \
```
Comment (v1v) [inline .ci/scripts/create-github-release.sh:22]: revert this?

```suggestion
  --repo "elastic/${REPO}" \
```

File: .ci/Jenkinsfile [hunk 7]
```diff
@@ -164,10 +165,8 @@ pipeline {
         BUCKET_SIGNED_ARTIFACTS_PATH = "gs://${env.BUCKET_NAME}/${env.BUCKET_SUBFOLDER_SIGNED_ARTIFACTS}"
         DIST_FOLDER = 'dist'
         RELEASE_URL_MESSAGE = "(<https://github.com/elastic/${env.REPO}/releases/tag/${env.TAG_NAME}|${env.TAG_NAME}>)"
+        SIGNER_PROJECT_NAME = 'Elastic_Synthetics_Recorder'
       }
-      // There are some environmental issues with some of the CI workers
-      // let's use the one we know it works.
-      //agent { label 'macos-code-signer' }
```
Comment (v1v) [inline .ci/Jenkinsfile:170]: Let's keep this comment to help with the reason

File: .ci/Jenkinsfile (hunk 5 above)
Comment (cachedout) [reply .ci/Jenkinsfile:326]: @v1v Are you sure this won't break the lookup down on L328?

Comment (cachedout) [reply .ci/Jenkinsfile:326]: Ah, derp. My fault. I didn't see that L328 was actually just an error message. Ignore me.

**Review by cachedout — APPROVED** [review]

**Review by v1v — COMMENTED** [review]
LGTM! 🚀

**Pull request merged**